Inheritance task

MKotlobay opened the pull request.

Commit:
Inheritance task

yim222 commented:
Hi I have passed on your coed. 
It seems good. 

But I suggest you to do some more tests on the object creation, and to do more overriding methods, and to show polymorphism, as we've did on the lesson end. 

See the code here 

https://github.com/lingar-studying/c-sharp-basics-lingar/blob/main/c-sharp-basics-lingar/inheritance/TestInheritance.cs




![image](https://github.com/MKotlobay/studying-c-sharp-mark-kotlobay/assets/16339682/2f951e3c-35e0-4de3-84be-df8e5350065e)

MKotlobay commented:
Done.

Added a function that updates the version of the product that the default was firstly 1, and then i updated it to 2

![Inhertance - update](https://github.com/MKotlobay/studying-c-sharp-mark-kotlobay/assets/131815349/85eadc3b-09ed-426b-b12c-d1aebeb60bc4)

Commit:
Updated - virtual/override function for versions of products
